### DISCUSSION TITLE
Locking Repositories after Submission Date

### DISCUSSION THREAD

**Discussion by dfuchss** [ls1intum/Artemis#2707] [Q&A] [answered]
Hi everyone,

I was wondering if the following behavior is intentional:

If a student tries to turn in an assignment there are, as far as I have understood, two possibilities:
1. Using the Online Editor
2. Using a VCS (in my case, I tested GitLab)

After the submission date passed, I've noticed two different behaviors:
1. The online editor is locked - no submission is possible (as I suspected)
2. **The student is still able to push to the VCS that triggers the CI Pipeline**

Is this behavior intentional, or could I have set something up incorrectly for my test environment, or should the repository be locked after the deadline passes?

**Comment by krusche** [comment, marked as answer]
The repository will only be locked if build and test after due date and/or manual assessment has been activated.

One issue might be, that the scheduling profile is missing in the active spring profiles of your Artemis server.

**Reply by dfuchss** [reply]
Thanks a lot for the fast response.
I know that the scheduling profile is not active in my current setting. I'll check whether this changes the behavior in my test environment.

**Reply by dfuchss** [reply]
Actually that works :) 
Thanks!
